Unable to complete the build in win10+VScode

win10
Visual Studio Code: Version: 1.95.1 (user setup)
cmake version 3.28.0-rc5
mingw64: I tried three versions and none of them worked
(1)gcc version 7.3.0 (x86_64-posix-seh-rev0, Built by MinGW-W64 project)
(2)gcc version 8.1.0 (x86_64-posix-seh-rev0, Built by MinGW-W64 project)
(3)gcc version 14.1.0 (x86_64-posix-seh-rev0, Built by MinGW-Builds project)

this is build log:
`
[proc] Executing command: "C:\Program Files\CMake\bin\cmake.EXE" --build d:/AgIsoVirtualTerminal/build --config Debug --target all -j 10 --
[build] [  2%] Built target AgISOVirtualTerminal_rc_lib
[build] [  7%] Built target Utility
[build] [ 49%] Built target Isobus
[build] [ 50%] Building CXX object _deps/can_stack-build/hardware_integration/CMakeFiles/HardwareIntegration.dir/src/sys_tec_windows_plugin.cpp.obj
[build] In file included from D:/AgIsoVirtualTerminal/build/_deps/can_stack-src/hardware_integration/include/isobus/hardware_integration/sys_tec_windows_plugin.hpp:23,
[build]                  from D:\AgIsoVirtualTerminal\build\_deps\can_stack-src\hardware_integration\src\sys_tec_windows_plugin.cpp:12:
[build] D:/AgIsoVirtualTerminal/build/_deps/can_stack-src/hardware_integration/include/isobus/hardware_integration/Usbcan32.h:787:1: warning: multi-line comment [-Wcomment]
[build]  //#define USBCAN_CALC_TIMEDIFF(old_time,new_time) \
[build]  ^
[build] D:\AgIsoVirtualTerminal\build\_deps\can_stack-src\hardware_integration\src\sys_tec_windows_plugin.cpp: In member function 'virtual void isobus::SysTecWindowsPlugin::open()':
[build] D:\AgIsoVirtualTerminal\build\_deps\can_stack-src\hardware_integration\src\sys_tec_windows_plugin.cpp:69:47: warning: missing initializer for member '_tUcanInitCanParam::m_bMode' [-Wmissing-field-initializers]
[build]      tUcanInitCanParam initCANParameters = { 0 };
[build]                                                ^
[build] D:\AgIsoVirtualTerminal\build\_deps\can_stack-src\hardware_integration\src\sys_tec_windows_plugin.cpp:69:47: warning: missing initializer for member '_tUcanInitCanParam::m_bBTR0' [-Wmissing-field-initializers]
[build] D:\AgIsoVirtualTerminal\build\_deps\can_stack-src\hardware_integration\src\sys_tec_windows_plugin.cpp:69:47: warning: missing initializer for member '_tUcanInitCanParam::m_bBTR1' [-Wmissing-field-initializers]
[build] D:\AgIsoVirtualTerminal\build\_deps\can_stack-src\hardware_integration\src\sys_tec_windows_plugin.cpp:69:47: warning: missing initializer for member '_tUcanInitCanParam::m_bOCR' [-Wmissing-field-initializers]
[build] D:\AgIsoVirtualTerminal\build\_deps\can_stack-src\hardware_integration\src\sys_tec_windows_plugin.cpp:69:47: warning: missing initializer for member '_tUcanInitCanParam::m_dwAMR' [-Wmissing-field-initializers]
[build] D:\AgIsoVirtualTerminal\build\_deps\can_stack-src\hardware_integration\src\sys_tec_windows_plugin.cpp:69:47: warning: missing initializer for member '_tUcanInitCanParam::m_dwACR' [-Wmissing-field-initializers]
[build] D:\AgIsoVirtualTerminal\build\_deps\can_stack-src\hardware_integration\src\sys_tec_windows_plugin.cpp:69:47: warning: missing initializer for member '_tUcanInitCanParam::m_dwBaudrate' [-Wmissing-field-initializers]
[build] D:\AgIsoVirtualTerminal\build\_deps\can_stack-src\hardware_integration\src\sys_tec_windows_plugin.cpp:69:47: warning: missing initializer for member '_tUcanInitCanParam::m_wNrOfRxBufferEntries' [-Wmissing-field-initializers]
[build] D:\AgIsoVirtualTerminal\build\_deps\can_stack-src\hardware_integration\src\sys_tec_windows_plugin.cpp:69:47: warning: missing initializer for member '_tUcanInitCanParam::m_wNrOfTxBufferEntries' [-Wmissing-field-initializers]
[build] D:\AgIsoVirtualTerminal\build\_deps\can_stack-src\hardware_integration\src\sys_tec_windows_plugin.cpp:70:40: warning: missing initializer for member '_tUcanChannelInfo::m_bMode' [-Wmissing-field-initializers]
[build]      tUcanChannelInfo channelInfo = { 0 };
[build]                                         ^
[build] D:\AgIsoVirtualTerminal\build\_deps\can_stack-src\hardware_integration\src\sys_tec_windows_plugin.cpp:70:40: warning: missing initializer for member '_tUcanChannelInfo::m_bBTR0' [-Wmissing-field-initializers]
[build] D:\AgIsoVirtualTerminal\build\_deps\can_stack-src\hardware_integration\src\sys_tec_windows_plugin.cpp:70:40: warning: missing initializer for member '_tUcanChannelInfo::m_bBTR1' [-Wmissing-field-initializers]
[build] D:\AgIsoVirtualTerminal\build\_deps\can_stack-src\hardware_integration\src\sys_tec_windows_plugin.cpp:70:40: warning: missing initializer for member '_tUcanChannelInfo::m_bOCR' [-Wmissing-field-initializers]
[build] D:\AgIsoVirtualTerminal\build\_deps\can_stack-src\hardware_integration\src\sys_tec_windows_plugin.cpp:70:40: warning: missing initializer for member '_tUcanChannelInfo::m_dwAMR' [-Wmissing-field-initializers]
[build] D:\AgIsoVirtualTerminal\build\_deps\can_stack-src\hardware_integration\src\sys_tec_windows_plugin.cpp:70:40: warning: missing initializer for member '_tUcanChannelInfo::m_dwACR' [-Wmissing-field-initializers]
[build] D:\AgIsoVirtualTerminal\build\_deps\can_stack-src\hardware_integration\src\sys_tec_windows_plugin.cpp:70:40: warning: missing initializer for member '_tUcanChannelInfo::m_dwBaudrate' [-Wmissing-field-initializers]
[build] D:\AgIsoVirtualTerminal\build\_deps\can_stack-src\hardware_integration\src\sys_tec_windows_plugin.cpp:70:40: warning: missing initializer for member '_tUcanChannelInfo::m_fCanIsInit' [-Wmissing-field-initializers]
[build] D:\AgIsoVirtualTerminal\build\_deps\can_stack-src\hardware_integration\src\sys_tec_windows_plugin.cpp:70:40: warning: missing initializer for member '_tUcanChannelInfo::m_wCanStatus' [-Wmissing-field-initializers]
[build] D:\AgIsoVirtualTerminal\build\_deps\can_stack-src\hardware_integration\src\sys_tec_windows_plugin.cpp:70:22: warning: variable 'channelInfo' set but not used [-Wunused-but-set-variable]
[build]      tUcanChannelInfo channelInfo = { 0 };
[build]                       ^~~~~~~~~~~
[build] D:\AgIsoVirtualTerminal\build\_deps\can_stack-src\hardware_integration\src\sys_tec_windows_plugin.cpp: In member function 'virtual bool isobus::SysTecWindowsPlugin::read_frame(isobus::CANMessageFrame&)':
[build] D:\AgIsoVirtualTerminal\build\_deps\can_stack-src\hardware_integration\src\sys_tec_windows_plugin.cpp:108:35: warning: missing initializer for member '_tCanMsgStruct::m_bFF' [-Wmissing-field-initializers]
[build]     tCanMsgStruct CANMessage = { 0 };
[build]                                    ^
[build] D:\AgIsoVirtualTerminal\build\_deps\can_stack-src\hardware_integration\src\sys_tec_windows_plugin.cpp:108:35: warning: missing initializer for member '_tCanMsgStruct::m_bDLC' [-Wmissing-field-initializers]
[build] D:\AgIsoVirtualTerminal\build\_deps\can_stack-src\hardware_integration\src\sys_tec_windows_plugin.cpp:108:35: warning: missing initializer for member '_tCanMsgStruct::m_bData' [-Wmissing-field-initializers]
[build] D:\AgIsoVirtualTerminal\build\_deps\can_stack-src\hardware_integration\src\sys_tec_windows_plugin.cpp:108:35: warning: missing initializer for member '_tCanMsgStruct::m_dwTime' [-Wmissing-field-initializers]
[build] D:\AgIsoVirtualTerminal\build\_deps\can_stack-src\hardware_integration\src\sys_tec_windows_plugin.cpp:130:10: error: 'std::this_thread' has not been declared
[build]      std::this_thread::sleep_for(std::chrono::milliseconds(1)); // This is long enough for 2 messages to be received at max bus load, which should be fine.
[build]           ^~~~~~~~~~~
[build] mingw32-make.exe[2]: *** [_deps\can_stack-build\hardware_integration\CMakeFiles\HardwareIntegration.dir\build.make:151: _deps/can_stack-build/hardware_integration/CMakeFiles/HardwareIntegration.dir/src/sys_tec_windows_plugin.cpp.obj] Error 1
[build] mingw32-make.exe[1]: *** [CMakeFiles\Makefile2:326: _deps/can_stack-build/hardware_integration/CMakeFiles/HardwareIntegration.dir/all] Error 2
[proc] The command: "C:\Program Files\CMake\bin\cmake.EXE" --build d:/AgIsoVirtualTerminal/build --config Debug --target all -j 10 -- exited with code: 2
[build] mingw32-make.exe: *** [Makefile:155: all] Error 2
[driver] Build completed: 00:00:02.608
[build] Build finished with exit code 2

`

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -13,7 +13,12 @@ set(BUILD_TESTING OFF)
 if(WIN32)
     set(CAN_DRIVER "WindowsPCANBasic")
     list(APPEND CAN_DRIVER "TouCAN")
-    list(APPEND CAN_DRIVER "WindowsInnoMakerUSB2CAN")
+
+    if(NOT (CMAKE_CXX_COMPILER_ID MATCHES "GNU" AND MINGW))
+      list(APPEND CAN_DRIVER "WindowsInnoMakerUSB2CAN")
+    else()
+      message(STATUS "InnoMaker hardware will not be supported in this build because you are using mingw")
+    endif()
     list(APPEND CAN_DRIVER "SYS_TEC")
 elseif(APPLE)
     set(CAN_DRIVER "MacCANPCAN")
@@ -91,7 +96,7 @@ target_link_libraries(
           isobus::Utility
   PUBLIC juce::juce_recommended_config_flags juce::juce_recommended_lto_flags)
 
-if(MSVC)
+if(WIN32)
     add_custom_command(
             TARGET AgISOVirtualTerminal POST_BUILD
             COMMAND "${CMAKE_COMMAND}" -E copy_if_different ${CMAKE_CURRENT_LIST_DIR}/lib/PCANBasic.dll "$<TARGET_FILE_DIR:AgISOVirtualTerminal>"

--- a/src/ConfigureHardwareComponent.cpp
+++ b/src/ConfigureHardwareComponent.cpp
@@ -28,13 +28,18 @@ ConfigureHardwareComponent::ConfigureHardwareComponent(ConfigureHardwareWindow &
 #ifdef JUCE_WINDOWS
 	hardwareInterfaceSelector.setName("Hardware Interface");
 	hardwareInterfaceSelector.setTextWhenNothingSelected("Select Hardware Interface");
-	hardwareInterfaceSelector.addItemList({ "PEAK PCAN USB", "Innomaker2CAN", "TouCAN", "SysTec" }, 1);
 
+#ifdef ISOBUS_WINDOWSINNOMAKERUSB2CAN_AVAILABLE
+	hardwareInterfaceSelector.addItemList({ "PEAK PCAN USB", "Innomaker2CAN", "TouCAN", "SysTec" }, 1);
+#else
+	hardwareInterfaceSelector.addItemList({ "PEAK PCAN USB", "Innomaker2CAN (not supported with mingw)", "TouCAN", "SysTec" }, 1);
+#endif
 	int selectedID = 1;
 
 	for (std::uint8_t i = 0; i < parentCANDrivers.size(); i++)
 	{
-		if (parentCANDrivers.at(i) == isobus::CANHardwareInterface::get_assigned_can_channel_frame_handler(0))
+		if ((nullptr != parentCANDrivers.at(i)) && 
+			(parentCANDrivers.at(i) == isobus::CANHardwareInterface::get_assigned_can_channel_frame_handler(0)))
 		{
 			selectedID = i + 1;
 			break;

--- a/src/Main.cpp
+++ b/src/Main.cpp
@@ -77,7 +77,11 @@ public:
 		{
 #ifdef JUCE_WINDOWS
 			canDrivers.push_back(std::make_shared<isobus::PCANBasicWindowsPlugin>(static_cast<WORD>(PCAN_USBBUS1)));
+#ifdef ISOBUS_WINDOWSINNOMAKERUSB2CAN_AVAILABLE
 			canDrivers.push_back(std::make_shared<isobus::InnoMakerUSB2CANWindowsPlugin>(0));
+#else
+			canDrivers.push_back(nullptr);
+#endif
 			canDrivers.push_back(std::make_shared<isobus::TouCANPlugin>(static_cast<std::int16_t>(0), 0));
 			canDrivers.push_back(std::make_shared<isobus::SysTecWindowsPlugin>());
 #elif defined(JUCE_MAC)

